scripts/py_matter_idl/matter_idl/backwards_compatibility/__init__.py

```diff
@@ -0,0 +1,292 @@
+# Copyright (c) 2022 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import dataclasses
+import enum
+import itertools
+import logging
+from typing import Callable, Dict, List, Optional, Protocol, TypeVar
+
+from matter_idl.matter_idl_types import Attribute, Bitmap, Cluster, ClusterSide, Command, Enum, Event, Field, Idl, Struct
+
+
+class Compatibility(enum.Enum):
+    UNKNOWN = enum.auto()
+    COMPATIBLE = enum.auto()
+    INCOMPATIBLE = enum.auto()
+
+
+T = TypeVar('T')
```

Comment:
This should use `typing.Any`  everywhere, rather than `T` which is a different concept.

Reply:
Not the same:

```
def group_list(items: List[T], get_id: Callable[[T], str]) -> Dict[str, T]:
```

says that you get a list of T and you get a dict with T

```
def group_list(items: List[Any], get_id: Callable[[T], str]) -> Dict[str, Any]:
```

does not say that the dict contains the same elements as the list.